scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -269,25 +449,41 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
           newlineStreak &= scannerTokens(i).is[Whitespace]
           i += 1
         }
+
         if (lastNewlinePos != -1 &&
           prev != null && prev.is[CanEndStat] &&
           next != null && next.isNot[CantStartStat] &&
-          (sepRegions.isEmpty || sepRegions.head == '}' || sepRegions.head == '$')) {
-          var token = scannerTokens(lastNewlinePos)
-          if (newlines) token = LFLF(token.input, token.dialect, token.start, token.end)
-          parserTokens += token
-          parserTokenPositions += lastNewlinePos
-          loop(lastNewlinePos, currPos + 1, sepRegions)
+          (sepRegionsParameter.isEmpty || sepRegionsParameter.head.closing == RegionBrace || sepRegionsParameter.head.closing == RegionEnum || sepRegionsParameter.head.closing == RegionIndent || sepRegionsParameter.head.closing == RegionIndentEnum)) {
+
+          if (isLeadingInfixOperator(next)) {
```

Comment:
What case is it solving?

Reply:
https://dotty.epfl.ch/docs/reference/other-new-features/indentation.html
`the first token on the next line is not a leading infix operator.`